diff --git a/src/others/file_explorer/lv_file_explorer.c b/src/others/file_explorer/lv_file_explorer.c
index 0563aaeef..7ce3f1743 100644
--- a/src/others/file_explorer/lv_file_explorer.c
+++ b/src/others/file_explorer/lv_file_explorer.c
@@ -13,7 +13,9 @@
 
 #include "../../lvgl.h"
 #include "../../core/lv_global.h"
+#include <ctype.h>
 
+#include "v_common.h"
 /*********************
  *      DEFINES
  *********************/
@@ -106,6 +108,9 @@ void lv_file_explorer_set_quick_access_path(lv_obj_t * obj, lv_file_explorer_dir
         case LV_EXPLORER_VIDEO_DIR:
             dir_str = &(explorer->video_dir);
             break;
+        case LV_EXPLORER_DOWNLOADS_DIR:
+            dir_str = &(explorer->downloads_dir);
+            break;
         case LV_EXPLORER_DOCS_DIR:
             dir_str = &(explorer->docs_dir);
             break;
@@ -309,8 +314,10 @@ static void lv_file_explorer_constructor(const lv_obj_class_t * class_p, lv_obj_
 
     btn = lv_list_add_button(explorer->list_places, NULL, LV_SYMBOL_HOME " HOME");
     lv_obj_add_event_cb(btn, quick_access_event_handler, LV_EVENT_CLICKED, obj);
-    btn = lv_list_add_button(explorer->list_places, NULL, LV_SYMBOL_VIDEO " Video");
+    btn = lv_list_add_button(explorer->list_places, NULL, LV_SYMBOL_DOWNLOAD " Downloads");
     lv_obj_add_event_cb(btn, quick_access_event_handler, LV_EVENT_CLICKED, obj);
+//    btn = lv_list_add_button(explorer->list_places, NULL, LV_SYMBOL_VIDEO " Video");
+//    lv_obj_add_event_cb(btn, quick_access_event_handler, LV_EVENT_CLICKED, obj);
     btn = lv_list_add_button(explorer->list_places, NULL, LV_SYMBOL_IMAGE " Pictures");
     lv_obj_add_event_cb(btn, quick_access_event_handler, LV_EVENT_CLICKED, obj);
     btn = lv_list_add_button(explorer->list_places, NULL, LV_SYMBOL_AUDIO " Music");
@@ -443,6 +450,9 @@ static void quick_access_event_handler(lv_event_t * e)
         if((lv_strcmp(label_text, LV_SYMBOL_HOME " HOME") == 0)) {
             path = &(explorer->home_dir);
         }
+        else if((lv_strcmp(label_text, LV_SYMBOL_DOWNLOAD " Downloads") == 0)) {
+            path = &(explorer->downloads_dir);
+        }
         else if((lv_strcmp(label_text, LV_SYMBOL_VIDEO " Video") == 0)) {
             path = &(explorer->video_dir);
         }
@@ -563,6 +573,44 @@ static void browser_file_event_handler(lv_event_t * e)
     }
 }
 
+#ifndef ARRAY_SIZE
+#define ARRAY_SIZE(a) (sizeof(a) / sizeof(a[0]))
+#endif
+
+const char *supported_ext[] = {
+    ".pdf",
+    ".musicxml",
+    ".mxl",
+    ".xml",
+    ".midi",
+    ".mid",
+    ".svg",
+};
+static bool supported_format(const char *file)
+{
+    int size = ARRAY_SIZE(supported_ext);
+    for (int i = 0; i < size; i++)
+    {
+        int ext_len = lv_strlen(supported_ext[i]);
+        int len = lv_strlen(file);
+        const char *file_ext = file + len - ext_len;
+        bool match = true;
+        for (int j = 0; j < ext_len; j++)
+        {
+            if ((uint8_t)tolower(file_ext[j]) != supported_ext[i][j])
+            {
+                match = false;
+                break;
+            }
+        }
+        if (match)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
 static void show_dir(lv_obj_t * obj, const char * path)
 {
     lv_file_explorer_t * explorer = (lv_file_explorer_t *)obj;
@@ -597,6 +645,11 @@ static void show_dir(lv_obj_t * obj, const char * path)
             LV_LOG_USER("No more files to read!");
             break;
         }
+#if HIDE_HIDDEN_FILES
+        if (fn[0] == '.' || (lv_strlen(fn) > 1 && fn[0] == '/' && fn[1] == '.')) {
+            continue;
+        }
+#endif
 
         file_entry_user_data = lv_malloc(sizeof(lv_file_explorer_file_table_entry_data_t));
         LV_ASSERT_MALLOC(file_entry_user_data);
@@ -626,6 +679,9 @@ static void show_dir(lv_obj_t * obj, const char * path)
             file_entry_user_data->file_kind = LV_FILE_EXPLORER_FILE_KIND_DIR;
         }
         else {
+            if (!supported_format(fn)) {
+                continue;
+            }
             lv_table_set_cell_value_fmt(explorer->file_table, index, 0, LV_SYMBOL_FILE "  %s", fn);
             file_entry_user_data->file_kind = LV_FILE_EXPLORER_FILE_KIND_FILE;
         }
@@ -639,13 +695,13 @@ static void show_dir(lv_obj_t * obj, const char * path)
 
     lv_table_set_row_count(explorer->file_table, index);
     file_explorer_sort(obj);
-    lv_obj_send_event(obj, LV_EVENT_READY, NULL);
+    lv_obj_send_event(obj, LV_EVENT_READY, &path[2]);
 
     /*Move the table to the top*/
     lv_obj_scroll_to_y(explorer->file_table, 0, LV_ANIM_OFF);
 
     lv_strncpy(explorer->current_path, path, sizeof(explorer->current_path));
-    lv_label_set_text_fmt(explorer->path_label, LV_SYMBOL_EYE_OPEN" %s", path);
+    lv_label_set_text_fmt(explorer->path_label, "%s", &path[2]);
 
     size_t current_path_len = lv_strlen(explorer->current_path);
     if((explorer->current_path[current_path_len - 1] != '/') && (current_path_len < LV_FILE_EXPLORER_PATH_MAX_LEN)) {
diff --git a/src/others/file_explorer/lv_file_explorer.h b/src/others/file_explorer/lv_file_explorer.h
index 818b8d85e..356689290 100644
--- a/src/others/file_explorer/lv_file_explorer.h
+++ b/src/others/file_explorer/lv_file_explorer.h
@@ -37,6 +37,7 @@ typedef enum {
     LV_EXPLORER_MUSIC_DIR,
     LV_EXPLORER_PICTURES_DIR,
     LV_EXPLORER_VIDEO_DIR,
+    LV_EXPLORER_DOWNLOADS_DIR,
     LV_EXPLORER_DOCS_DIR,
     LV_EXPLORER_FS_DIR,
 } lv_file_explorer_dir_t;
diff --git a/src/others/file_explorer/lv_file_explorer_private.h b/src/others/file_explorer/lv_file_explorer_private.h
index ea35dd288..890a1841e 100644
--- a/src/others/file_explorer/lv_file_explorer_private.h
+++ b/src/others/file_explorer/lv_file_explorer_private.h
@@ -43,6 +43,7 @@ struct _lv_file_explorer_t {
     char * music_dir;
     char * pictures_dir;
     char * video_dir;
+    char * downloads_dir;
     char * docs_dir;
     char * fs_dir;
 #endif
